app/src/main/java/com/ubergeek42/WeechatAndroid/utils/History.kt

```diff
@@ -0,0 +1,54 @@
+package com.ubergeek42.WeechatAndroid.utils
+
+import android.os.Bundle
+import android.text.Editable
+import android.widget.EditText
+import androidx.core.os.bundleOf
+import com.ubergeek42.WeechatAndroid.service.P
+
+class History {
+    private var index = -1
+    private var userInput: Editable? = null
+    private var selectionStart: Int? = null
+    private var selectionEnd: Int? = null
+
+    fun navigateOffset(editText: EditText, offset: Int) {
```

Comment:
Not a fan of the function name, needs to reflect what it does in a simpler way. Also I'd make it only handle +1 or -1 offsets for simplicity. I guess I'd use an enum

Reply:
Used an enum instead. I can't think of something better than "navigate", please suggest a name if you still don't like it.